src/services/rfqm_service.ts

```diff
@@ -513,6 +528,184 @@ export class RfqmService {
         return;
     }
 
+    /**
+     * Submit a transaction, re-submit if necessary, and poll for resolution
+     */
+    public async completeSubmissionLifecycleAsync(
+        orderHash: string,
+        workerAddress: string,
+        callData: string,
+    ): Promise<void> {
+        let submissionsMap: SubmissionsMap = {};
+
+        let gasPrice: BigNumber;
+        let nonce: number;
+        let gasEstimate: number;
+
+        [gasPrice, nonce, gasEstimate] = await Promise.all([
+            this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync(),
+            this._blockchainUtils.getNonceAsync(workerAddress),
+            this._blockchainUtils.estimateGasForExchangeProxyCallAsync(callData, workerAddress),
+        ]);
+
+        const expectedTakerTokenFillAmount = this._blockchainUtils.getTakerTokenFillAmountFromMetaTxCallData(callData);
+
+        const firstSubmission = await this._submitTransactionAsync(
+            orderHash,
+            workerAddress,
+            callData,
+            gasPrice,
+            nonce,
+            gasEstimate,
+            expectedTakerTokenFillAmount,
+        );
+        submissionsMap[firstSubmission.transactionHash!] = firstSubmission;
+
```

Comment:
Offering a small suggestion for refactor, unnecessary to do if you don't feel like it.

```
        let latestStatusCheck: SubmissionsMapStatus = {
            isTxFinalized: false,
            isTxMined: false,
            submissionsMap: {
                [firstSubmission.transactionHash!]: firstSubmission,
            }
        };
        while (!latestStatusCheck.isTxFinalized) {
            await delay(TRANSACTION_WATCHER_SLEEP_TIME_MS);

            latestStatusCheck = await this._checkSubmissionMapReceiptsAndUpdateDbAsync(
                latestStatusCheck.submissionsMap,
                expectedTakerTokenFillAmount,
            );
            if (!latestStatusCheck.isTxMined) {
                const newGasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();

                if (gasPrice.lt(newGasPrice)) {
                    gasPrice = newGasPrice;
                    const submission = await this._submitTransactionAsync(
                        orderHash,
                        workerAddress,
                        callData,
                        gasPrice,
                        nonce,
                        gasEstimate,
                        expectedTakerTokenFillAmount,
                    );
                    latestStatusCheck.submissionsMap[submission.transactionHash!] = submission;
                }
            }
        }
    }
```

Reply:
This allows you to avoid the
```
isTxFinalized = false
isTxMined = false
```